src/core/tileset.cpp

```diff
@@ -132,73 +126,152 @@ QList<QRgb> Tileset::getPalette(int paletteId, Tileset *primaryTileset, Tileset
+        dataString.append(QString("\t.4byte gMetatiles_%1\n").arg(friendlyName));
+        if (projectConfig.getBaseGameVersion() == BaseGameVersion::pokefirered) {
+            dataString.append("\t.4byte NULL @ animation callback\n");
+            dataString.append(QString("\t.4byte gMetatileAttributes_%1\n").arg(friendlyName));
+        } else {
+            dataString.append(QString("\t.4byte gMetatileAttributes_%1\n").arg(friendlyName));
+            dataString.append("\t.4byte NULL @ animation callback\n");
+        }
     } else {
-        dataString.append(QString("\t.4byte gMetatileAttributes_%1\n").arg(friendlyName));
-        dataString.append("\t.4byte NULL\n");
+        // Append to C file
+        dataString.append(QString("const struct Tileset %1 =\n{\n").arg(this->name));
+        if (projectConfig.getTilesetsHaveIsCompressed()) dataString.append("    .isCompressed = TRUE,\n");
+        dataString.append(QString("    .isSecondary = %1,\n").arg(this->is_secondary));
+        dataString.append(QString("    .tiles = gTilesetTiles_%1,\n").arg(friendlyName));
+        dataString.append(QString("    .palettes = gTilesetPalettes_%1,\n").arg(friendlyName));
+        dataString.append(QString("    .metatiles = gMetatiles_%1,\n").arg(friendlyName));
+        dataString.append(QString("    .metatileAttributes = gMetatileAttributes_%1,\n").arg(friendlyName));
+        if (projectConfig.getTilesetsHaveCallback()) dataString.append("    .callback = NULL,\n");
+        dataString.append("};\n");
     }
     file.write(dataString.toUtf8());
     file.flush();
     file.close();
     return true;
 }
 
-bool Tileset::appendToGraphics(QString graphicsFile, QString friendlyName, bool primary) {
-    QString primaryString = primary ? "primary" : "secondary";
-    QString basePath = projectConfig.getFilePath(ProjectFilePath::data_tilesets_folders);
+bool Tileset::appendToGraphics(QString root, QString friendlyName, bool usingAsm) {
+    QString graphicsFile = root + "/" + (usingAsm ? projectConfig.getFilePath(ProjectFilePath::tilesets_graphics_asm)
+                                                  : projectConfig.getFilePath(ProjectFilePath::tilesets_graphics));
     QFile file(graphicsFile);
     if (!file.open(QIODevice::WriteOnly | QIODevice::Append)) {
         logError(QString("Could not write to file \"%1\"").arg(graphicsFile));
         return false;
     }
-    QString dataString = "\n\t.align 2\n";
-    dataString.append(QString("gTilesetPalettes_%1::\n").arg(friendlyName));
-    for(int i = 0; i < Project::getNumPalettesTotal(); ++i) {
-        QString paletteString = QString("%1.gbapal").arg(i, 2, 10, QLatin1Char('0'));
-        dataString.append(QString("\t.incbin \"%1%2/%3/palettes/%4\"\n").arg(basePath, primaryString, friendlyName.toLower(), paletteString));
 
+    const QString tilesetDir = this->getExpectedDir();
+    const QString tilesPath = tilesetDir + "/tiles.4bpp.lz";
```

Comment:
I think this tiles filepath needs to check the compressed config option.

Reply:
Porymap (currently at least) always outputs tileset paths compressed. We can change that if you want, idk how that would be presented in the UI. I don't think we should assume if `isCompressed` isn't being outputted that the user doesn't want it compressed

Reply:
I'm confused--isn't that exactly what the `isCompressed` option is used for?  It denotes if new tilesets have a compressed tiles image?

Reply:
Porymap doesn't care if the image will be compressed, it just needs to know where the image is (and the image is always `.png`). Porymap also never edits existing tileset headers. When outputting a new tileset header, currently and in this PR it will always set `isCompressed` to `TRUE` and output a `.lz` file path. The config option is just to stop Porymap from outputting the `isCompressed` field if e.g. a user removed it from their struct

Reply:
Ok, I'm with you.  I was interpreting the `isCompressed` config option to mean "I don't want my tiles image to be compressed", rather than "compressed is not part of my tileset struct at all".

Reply:
Gotcha, I realize now I forgot to update the manual for those new config options (which should clear that confusion up)